### ISSUE TITLE
Bug in CombinedHighlighter

### ISSUE THREAD

**Issue by vit001** [PhilJay/MPAndroidChart#1539]
```
        for ( int i = 0; i < dataObjects.size(); i++ ) {

            for ( int j = 0; j < dataObjects.get(i).getDataSetCount(); j++ ) {

                IDataSet dataSet = dataObjects.get(i).getDataSetByIndex(j);

                // dont include datasets that cannot be highlighted
                if ( ! dataSet.isHighlightEnabled() )
                    continue;

                // extract all y-values from all DataSets at the given x-index
                final float yVal = dataSet.getYValForXIndex( xIndex );
                if ( yVal == Float.NaN )
                    continue;

                pts[1] = yVal;

                mChart.getTransformer(dataSet.getAxisDependency()).pointValuesToPixel(pts);

                if ( ! Float.isNaN(pts[1]) ) {
                    vals.add(new SelectionDetail(pts[1], j, dataSet));
                }
            }
        }
```

should read

```
        for ( int i = 0; i < dataObjects.size(); i++ ) {

            for ( int j = 0; j < dataObjects.get(i).getDataSetCount(); j++ ) {

                IDataSet dataSet = dataObjects.get(i).getDataSetByIndex(j);

                // dont include datasets that cannot be highlighted
                if ( ! dataSet.isHighlightEnabled() )
                    continue;

                // extract all y-values from all DataSets at the given x-index
                final float yVal = dataSet.getYValForXIndex( xIndex );
                if ( yVal == Float.NaN )
                    continue;

                pts[1] = yVal;

                mChart.getTransformer(dataSet.getAxisDependency()).pointValuesToPixel(pts);

                if ( ! Float.isNaN(pts[1]) ) {
                    vals.add(new SelectionDetail(pts[1], i, dataSet)); // NOTE BUG HERE
                }
            }
        }
```

**Comment by vit001** [comment]
OK, it after further investigation, it appears the problem in this code is different. The issue is that for a CombinedChart, SelectionDetail needs to contain additional information to correctly/uniquely identify what to highlight: needs to store both i (which data type is being highlighted, e.g., ScatterData, LineData), AND j (which dataset is being highlighted, e.g. second ScatterDataSet). Also, this information needs to be used correctly in Chart.drawMarkers.

**Comment by PhilJay** [comment]
This should be fixed in v3.0.0
